### DISCUSSION TITLE
PreventDefault not working in next 13

### DISCUSSION THREAD

**Discussion by bash-codify** [vercel/next.js#55136] [Help]
### Summary

When using the e.preventDefault to prevent the default event behavior for a form submit, the page keeps refreshing.
<img width="454" alt="Screenshot 2023-09-08 025956" src="https://github.com/vercel/next.js/assets/105221850/191a1fc9-8cba-4ae2-b076-a1f9affa5b43">

**Comment by icyJoseph** [comment]
I cannot reproduce with the latest next version. 

Could you share some more info about your setup? or a link to a repository?

**Reply by bash-codify** [reply]
hello Joseph

I created a test repo and I'm facing the same issue

https://github.com/bash-codify/Test

**Reply by icyJoseph** [reply]
Hi, 

Unfortunately that's just a typo, https://github.com/bash-codify/Test/blob/main/src/Components/SearchForm.jsx#L16:

```jsx
  e.prvenetDefault();
```

You mean, `e.preventDefault()`

**Reply by bash-codify** [reply]
Hi J,

Thanks for your time.

Here is a main project and I didn't make any typo error here or maybe I didn't but I can't see it.

https://github.com/bash-codify/Aiirolafin-main/blob/master/src/Components/Banner/SearchForm.jsx

**Reply by icyJoseph** [reply]
You have:

```jsx
  <form onSubmit={`${(e)=> submitHandler(e)}`}> 
```

That's casting your handler to a string.

You want:

```tsx
  <form onSubmit={(e)=> submitHandler(e)}> 
```

or

```tsx
  <form onSubmit={submitHandler}> 
```

**Reply by bash-codify** [reply]
Have used both of these <form onSubmit={(e)=> submitHandler(e)}> and <form onSubmit={submitHandler}>

have corrected it, still the same issue.

**Reply by icyJoseph** [reply]
Could it be that Banner, which uses SearchForm, is an async component, but it uses global context, which is a client tree kind of thing?

**Reply by bash-codify** [reply]
i have remove the SearchForm from Banner component, but still the same.

**Reply by icyJoseph** [reply]
Well, I can almost guarantee that this is an issue within your code base. I see that the Banner is an async function, but it uses `use client` hook.

**Reply by icyJoseph** [reply]
And the Banner is also being rendered as part of a client tree... that's not good either, Banner can't be async in that case.

**Comment by borisboskovic** [comment]
I'm not sure if this is relevant, but here is something I noticed (Used Next 15, happens in __DEV mode only__):

If you try to submit immediately after reloading the page, `preventDefault` will not work and the page will refresh. I don't know if that's due to hydration not being completed or if it's expected. When the page loads completely, `preventDefault` will work.

It happens even with the most basic example.

```tsx
"use client"

import {FormEventHandler} from "react";

export default function Home() {
    const submitHandler: FormEventHandler<HTMLFormElement> = (event) => {
        event.preventDefault()
        console.log("Submit")
    }

    return (
        <div>
            <form onSubmit={submitHandler}>
                <input type="text" name="text" />
                <button type="submit">Submit</button>
            </form>
        </div>
    );
}
```
